### DISCUSSION TITLE
Multiple project config folders/paths with the same content

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#9122] [Q&A] [answered]
What's the purpose of multiple different cryptic project config folders with the same content?
I tried to dig into the source Code but I wasn't able to find a reference for it.
Deleting the duplicates and re-building the config didn't work (the files were re-created sooner or later) and I have this in multiple projects for > Craft 3.5.

What I mean is the following
![image](https://user-images.githubusercontent.com/6122122/124482707-deaca500-dda9-11eb-9c49-a69a41f96f29.png)

There are 3 folders with the same content for global sets und 4 folders that contain the entire fields config. These files are the same - no differences at all.
Are those meant to be temp files before or after an update?
I would just like to know their purpose.

**Comment by andris-sevcenko** [comment, marked as answer]
That is not something Craft does. At least intentionally?

Can you check the contents of your `projectconfig` table and see if anything jumps at you?

**Reply by Anubarak** [reply]
That's what I already did 6 months ago and I wasn't able to find anything related to those paths.  The table seems totally fine, there are no duplicate records in there (in none of the projects where the issue occurs) and no other issues. Everything works fine that's why I didn't mention it earlier but it always bothered me.

I even tried to find relations between those strange folder names and uids or handles but as far as I can see there are none.

**Reply by Anubarak** [reply]
Maybe those are meant to be for deleted things?
I deleted a global set recently -> the set was removed from project config
I run project-config/rebuild a few minutes ago now the project config file for this set was re-created but in the 
`..cDi` folder (not in the normal globalSets folder)
it now looks like this

![image](https://user-images.githubusercontent.com/6122122/124485247-ad81a400-ddac-11eb-85f6-1db53f6eeb4d.png)
The set is still marked as "deleted" tho + my projectconfig table does not contain values from this file

**Reply by andris-sevcenko** [reply]
Can you try clearing your caches and then running `project-config/rebuild`?

**Reply by Anubarak** [reply]
Sorry for the late response..

I deleted those file, rebuild the config and they were re-created
after that I cleared all caches, deleted those files, rebuild the config, this time they were **not** re-created

I guess I know what I have to do for my other projects.
Thank you very much for your help

**Reply by brandonkelly** [reply]
Very weird…

**Reply by Anubarak** [reply]
@brandonkelly I think I know when it happens, however I don't have the time to figure out why it happens.
Our internal server is a little bit "misconfigured" to say at least.. we cannot run a composer update nor craft commands without root permissions (don't judge me, I have nothing to do with it) but doing so via root will also change the permissions of config files that were touched via migrations.

When I forget to change all the permissions back to the web user I'll receive an exception "config isn't writable" - after changing all permissions and re-save something all those strange folders appear until I delete from and delete the cache.
Not sure if this is really an "issue" or if you should spend your time to fix it unless it's trivial to do so.
